core/src/services/d1/backend.rs

```diff
@@ -0,0 +1,376 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+
+use async_trait::async_trait;
+use http::header;
+use http::Request;
+use http::StatusCode;
+use serde_json::{de, Value};
+
+use crate::raw::adapters::kv;
+use crate::raw::*;
+use crate::*;
+
+use super::error::parse_error;
+use super::model::D1Response;
+
+#[doc = include_str!("docs.md")]
+#[derive(Default)]
+pub struct D1Builder {
+    token: Option<String>,
+    account_identifier: Option<String>,
+    database_identifier: Option<String>,
+
+    endpoint: Option<String>,
+    http_client: Option<HttpClient>,
+    root: Option<String>,
+
+    table: Option<String>,
+    key_field: Option<String>,
+    value_field: Option<String>,
+}
+
+impl Debug for D1Builder {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("D1Builder");
+        ds.field("endpoint", &self.endpoint);
+        ds.field("root", &self.root);
+        ds.field("table", &self.table);
+        ds.field("key_field", &self.key_field);
+        ds.field("value_field", &self.value_field);
+        ds.finish()
+    }
+}
+
+impl D1Builder {
+    /// Set api token for the cloudflare d1 service.
+    ///
+    /// create a api token from [here](https://dash.cloudflare.com/profile/api-tokens)
+    pub fn token(&mut self, token: &str) -> &mut Self {
+        if !token.is_empty() {
+            self.token = Some(token.to_string());
+        }
+        self
+    }
+
+    /// Set the account identifier for the cloudflare d1 service.
+    ///
+    /// get the account identifier from Workers & Pages -> Overview -> Account ID
+    /// default: "account-identifier"
+    pub fn account_identifier(&mut self, account_identifier: &str) -> &mut Self {
+        if !account_identifier.is_empty() {
+            self.account_identifier = Some(account_identifier.to_string());
+        }
+        self
+    }
+
+    /// Set the database identifier for the cloudflare d1 service.
+    ///
+    /// get the database identifier from Workers & Pages -> D1 -> [Your Database] -> Database ID
+    /// default: "database-identifier"
+    pub fn database_identifier(&mut self, database_identifier: &str) -> &mut Self {
+        if !database_identifier.is_empty() {
+            self.database_identifier = Some(database_identifier.to_string());
+        }
+        self
+    }
+
+    /// Set endpoint for http backend.
+    ///
+    /// For more information, please refer to [D1 Database API](https://developers.cloudflare.com/api/operations/cloudflare-d1-query-database)
+    /// default: "https://api.cloudflare.com/client/v4"
+    pub fn endpoint(&mut self, v: &str) -> &mut Self {
```

Comment:
We don't need to specify the endpoint. There is only one instance of cloudflare.